features/enable_fips_cloud.feature

```diff
@@ -617,3 +648,70 @@ Feature: FIPS enablement in cloud based machines
         Examples: ubuntu release
            | release | fips-name    | fips-service |fips-apt-source                                |
            | bionic  | FIPS Updates | fips-updates |https://esm.ubuntu.com/fips-updates/ubuntu bionic-updates/main |
+
+    @slow
+    @series.focal
+    @uses.config.machine_type.gcp.generic
```

Comment:
For GCP we are also testing for `fips-updates`. Should we add that same test procedure for aws and azure ?
Also, the main difference here between fips and fips-updates is the hold logic we test on FIPS. Do you think we should keep that separation or remove that part of the fips test and unify the fips and fips updates scenarios ?  PS: We would still have the hold test on the VMs.

What do you think @renanrodrigo @orndorffgrant ?

Reply:
I'd say yes, it is important to test fips-updates for Azure/AWS as well, and I think those should be separate tests as they are for GCP. Justification for both is that `fips-updates` is considered a separate service from the UA perspective.

Reply:
I think we should have tests for fips-updates for all the clouds, however I'm wondering if theres a decent way to avoid too much repetition in our test code for this.
Could we move the `apt hold` test steps to their own scenario, and reuse the other fips scenarios for fips and fips-updates.

Maybe something like this for the apt hold test:
```
    @series.bionic
    @series.focal
    @uses.config.machine_type.aws.generic
    @uses.config.machine_type.azure.generic
    @uses.config.machine_type.gcp.generic
    Scenario Outline: FIPS unholds packages
        Given a `<release>` machine with ubuntu-advantage-tools installed
        When I attach `contract_token` with sudo
        And I run `DEBIAN_FRONTEND=noninteractive apt-get install -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" -y openssh-client openssh-server strongswan` with sudo
        And I run `apt-mark hold openssh-client openssh-server strongswan` with sudo
        And I run `ua enable fips --assume-yes` with sudo
        When I run `ua disable fips --assume-yes` with sudo
        When I run `apt-mark unhold openssh-client openssh-server strongswan` with sudo
        Then I will see the following on stdout:
        """
        openssh-client was already not hold.
        openssh-server was already not hold.
        strongswan was already not hold.
        """
        Examples: ubuntu release
           | release |
           | bionic  |
           | focal   |
```

Then the apt hold logic could be removed from the other fips tests and they could be re-used for fips and fips-updates via the "Examples" table

What do you think @lucasmoura @renanrodrigo ?

Reply:
I think that is a great idea

Reply:
Just re-looked at what we are testing, and yes, I do think it works

Reply:
I will perform that change then

Reply:
@CalvoM  @orndorffgrant @renanrodrigo I have updated the tests, but I made some extra modifications there. Let me know what you all think